### PR TITLE
[RUSH] Filter

### PR DESCRIPTION
**Pull request by DhanBirSunwar**
This PR includes category filter and date range filter.

### CODE REVIEW

File: src/components/expenses/expense-listing/list-filter.tsx [hunk 1]
```diff
@@ -6,32 +7,72 @@ import {
...
+import { getCategory } from "../../providers/database/category";
+import { useQueryState } from "nuqs";
+import { Button } from "../../ui/button";
 
 const ExpenseListFilter = () => {
+  const [catagories, setCategories] = useState<Category[]>([]);
+  const [category, setCategory] = useQueryState("category");
+  const [start, setStart] = useQueryState("start");
+  const [end, setEnd] = useQueryState("end");
+  const res = async () => {
```
Comment (tashi-iu) [inline src/components/expenses/expense-listing/list-filter.tsx:21]: Proper name, maybe `fetchCategories`

File: src/components/expenses/expense-listing/list-filter.tsx [hunk 2]
```diff
@@ -6,32 +7,72 @@ import {
...
+    await getCategory().then((res) => setCategories(res?.categories));
+  };
+
+  useEffect(() => {
+    res();
+  }, []);
+
+  const handleclear = () => {
```
Comment (tashi-iu) [inline src/components/expenses/expense-listing/list-filter.tsx:29]: `handleClear`

File: src/components/expenses/expense-listing/list-table.tsx [hunk 3]
```diff
@@ -74,8 +79,20 @@ const ExpenseListTable = () => {
       setloading(false);
     });
 
+  const filteredList = useMemo(() => {
+    const filterCategory = category
+      ? tablelist.filter((expense) => expense.category == category)
+      : tablelist;
+    return filterCategory.filter((expense) => {
```
Comment (tashi-iu) [inline src/components/expenses/expense-listing/list-table.tsx:86]:
```ts
    return filterCategory.filter((expense) => 
        !start || !end || isWithinInterval(expense.date, { start, end })
    );
```

File: src/components/expenses/expense-listing/list-table.tsx [hunk 4]
```diff
@@ -86,17 +103,20 @@ const ExpenseListTable = () => {
       columnVisibility,
     },
   });
+
   if (loading) return <p>Loading...</p>;
 
+  console.log("rendering");
```
Comment (tashi-iu) [inline src/components/expenses/expense-listing/list-table.tsx:109]: Remove

**Review by tashi-iu — APPROVED** [review]

**Review by dorji-dev — APPROVED** [review]

**Pull request merged**